### PR TITLE
Feature/product detail page

### PR DESCRIPTION
**Pull request by mmoebeck**

### CODE REVIEW

**Review by jflheureux — CHANGES_REQUESTED** [review]
As the code will change a lot by extracting the component(s), I will review the logic and rest of the code when the major changes are done.

**Comment by mmoebeck** [comment]
I have pushed up changes.

File: Website/src/rendering/src/stories/Products/ProductDetailPage.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export default {
+  title: 'Components/Products/ProductDetailPage',
+  component: ProductDetailPage,
+} as ComponentMeta<typeof ProductDetailPage>;
+
+const Template: ComponentStory<typeof ProductDetailPage> = (args) => <ProductDetailPage {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
```
Comment (jflheureux) [inline Website/src/rendering/src/stories/Products/ProductDetailPage.stories.tsx:14]: Now that the cart PR with OrderCloud Redux store mock is merged in `feature/reflektion-poc`, could you please:
1. Merge `feature/reflektion-poc` into your branch.
2. Fix merge conflicts, if any,
3. Implement Storybook variants for the various states/possibilities of this component.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            <div className="product-description">
+              <h2>{product.Name}</h2>
+              <p>SKU: {sku}</p>
+              <div>{product.Description}</div>
+              <div>
+                Price:{' '}
+                <span className="product-price">{product.PriceSchedule.PriceBreaks[0].Price}</span>
+              </div>
+              <form onSubmit={handleAddToCart}>
+                {specs ? (
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:259]: My previous comment about avoiding to mix markup and logic in the main return statement was not just for the productImage. I also mentioned to please extract constants for all the parts that are conditional in the pull request code.

This line is another example of a condition inside the main JSX of the component.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+                            onChange={handleSpecFieldChange}
+                            optionId={specValue && specValue.OptionID}
+                            value={specValue && specValue.Value}
+                          />
+                        );
+                      })}
+                    </div>
+                  </div>
+                ) : null}
+                {variants && !variantID ? (
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:280]: This line is another example of a condition inside the main JSX of the component.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+                Price:{' '}
+                <span className="product-price">{product.PriceSchedule.PriceBreaks[0].Price}</span>
+              </div>
+              <form onSubmit={handleAddToCart}>
+                {specs ? (
+                  <div>
+                    Specs:
+                    <div className="product-specs">
+                      {specs?.map((spec) => {
+                        const specValue = specValues.find(
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:264]: Would it be possible to move this lookup logic into a separate function outside the main JSX return to help keep the JSX as logic-free as possible? Thanks

File: Website/src/rendering/src/components/Products/ProductQuantityInput.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    onChange(Number(e.target.value));
+  };
+
+  return (
+    <label htmlFor={controlId}>
+      {label}
+      {priceSchedule.RestrictedQuantity ? (
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductQuantityInput.tsx:32]: This line is another example of a condition inside the main JSX of the component.

File: Website/src/rendering/src/components/Products/ProductSpecField.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+      OptionID: 'OpenText',
+      Value: e.target.value.length ? e.target.value : undefined,
+    });
+  };
+
+  return (
+    <div>
+      <label htmlFor={spec.ID}>
+        {spec.Name}
+        {spec.OptionCount ? (
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductSpecField.tsx:37]: This line is another example of a condition inside the main JSX of the component.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+  batch: [Widget];
+};
+
+type specProps = {
+  SpecID: string;
+  OptionID?: string;
+  Value?: string;
+};
+
+const ProductDetailPage = (): JSX.Element => {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:41]: As the component is not really a page, maybe `ProductDetails` would be a name that better describes it. Could it please be renamed along with the Storybook file?

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+      value: [ProductProps];
+    };
+  };
+};
+
+type rfkResponse = {
+  batch: [Widget];
+};
+
+type specProps = {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:35]: `props` in react is a term to describe the input properties of a component. Here, I believe this type is not used as the input of a component. `Props` could be removed from its name.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+    rfkid: string;
+  };
+  content: {
+    product: {
+      value: [ProductProps];
+    };
+  };
+};
+
+type rfkResponse = {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:31]: Type and interface names should all start with a capital letter. This also applies to `specProps`.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+import { FormEvent, useCallback, useEffect, useState } from 'react';
+import useOcProductDetail from 'src/hooks/useOcProductDetail';
+import { createLineItem } from 'src/redux/ocCurrentCart';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { logViewEvent } from 'src/services/CdpService';
+import ProductQuantityInput from './ProductQuantityInput';
+import ProductSpecField from './ProductSpecField';
+import { Product } from './Shop';
+
+type ProductProps = {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:12]: `props` in react is a term to describe the input properties of a component. Here, I believe this type is not used as the input of a component. `Props` could be removed from its name.

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx (hunk 9 above)
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:31]: If this is not specific to the product details component, could it please be moved to `DiscoverService`?

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+  name: string;
+  price: number;
+  image_url: string;
+  brand: string;
+  sku: string;
+};
+
+type Widget = {
```
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:20]: If this is not specific to the product details component, could it please be moved to `DiscoverService`?

File: Website/src/rendering/src/components/Products/ProductDetailPage.tsx (hunk 10 above)
Comment (jflheureux) [inline Website/src/rendering/src/components/Products/ProductDetailPage.tsx:12]: If this is not specific to the product details component, could it please be moved to `DiscoverService`?

**Review by jflheureux — CHANGES_REQUESTED** [review]
Thanks for the previous changes. The code architecture is better now!

This is a partial review. I am not done reviewing all logic. At least these comments will allow you to improve the code while I continue to review the rest of the code.

**Review by jflheureux — APPROVED** [review]

**Pull request merged**